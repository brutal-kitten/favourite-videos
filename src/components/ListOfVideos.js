import React, { Component } from 'react';
import VideoContainer from './VideoContainer';
import NavButtons from './NavButtons';


class ListOfVideos extends Component {



  render() {

    return (
      <div className="list">
        <h3>My list</h3>
        <NavButtons
          showDemo={this.props.showDemo}
          showFavorite={this.props.showFavorite}
          deleteList={this.props.deleteList}
          sort={this.props.sort}
          showFav={this.props.showFav}
          returnToList={this.props.returnToList}
        />
        {(this.props.showFav ? (this.props.list.filter(item => (item.favorite === true))) : this.props.list).map((item) => (
        <VideoContainer
          key={item.id}
          id={item.id}
          title={item.title}
          date={item.date}
          views={item.views}
          likes={item.likes}
          thumbnails={item.thumbnails}
          isfavorite={item.favorite}
          removeFromFavorite={this.props.removeFromFavorite}
          addToFavorite={this.props.addToFavorite}
          deleteVideo={this.props.deleteVideo}
          playVideo={this.props.playVideo}
          changeSearchResultError={this.props.changeSearchResultError}
        />
      ))}
      </div>
    )
  }
}

export default ListOfVideos;
