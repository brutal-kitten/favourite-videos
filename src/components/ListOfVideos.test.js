import React from 'react';
import ReactDOM from 'react-dom';
import ListOfVideos from './ListOfVideos.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const props = {
  list: [
    {id: "hY7m5jjJ9mM", favorite: false, title: "CATS will make you LAUGH YOUR HEAD OFF - Funny CAT compilation" , date: new Date("2017-05-31T09:30:02.000Z").toDateString() , likes: "327046" , views: "81007082" , thumbnails: "https://i.ytimg.com/vi/hY7m5jjJ9mM/default.jpg" },
    {id: "mRf3-JkwqfU", favorite: false, title: "Funny Puppies And Cute Puppy Videos Compilation 2016 [BEST OF]" , date: new Date("2016-08-31T17:34:50.000Z").toDateString() , likes: "71389" , views: "17981823" , thumbnails: "https://i.ytimg.com/vi/mRf3-JkwqfU/default.jpg"}
  ]
}

it('renders properly', () => {
  const tree = TestRenderer.create(
    <ListOfVideos {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});


const wrapper = mount(<ListOfVideos {...props} />);

it('should have  header "my list"', () => {
  expect(wrapper.find('h3').exists()).toBeTruthy();
  expect(wrapper.find('h3').text()).toBe('My list');
});

it('should have rendered VideoContainer as much time as length of list', () => {
  expect(wrapper.find('.videocontainer').length).toBe(props.list.length);
})
