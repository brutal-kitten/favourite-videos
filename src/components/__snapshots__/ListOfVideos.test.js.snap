// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="list"
>
  <h3>
    My list
  </h3>
  <div
    className="navButtons grid-container2"
  >
    <div
      className="showDemo"
    >
      <button
        className="buttonShowDemo"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        Show demolist
      </button>
    </div>
    <div
      className="deleteList"
    >
      <button
        className="buttonDeleteList"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        Delete list
      </button>
    </div>
    <div
      className="showFav"
    >
      <button
        className="buttonShowFav"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        Show favorite
      </button>
    </div>
    <div
      className="sort"
    >
      <span
        className="sortText"
      >
        Sort by
      </span>
      <select
        id="select"
        onChange={[Function]}
        tabIndex="0"
        value="new"
      >
        <option
          value="new"
        >
          Newest added
        </option>
        <option
          value="old"
        >
          Oldest added
        </option>
      </select>
    </div>
  </div>
  <div
    className="listContainer"
  >
    <div
      className="grid-container videocontainer"
    >
      <div
        className="pic grid-item"
      >
        <img
          alt="thumbnail"
          onClick={[Function]}
          src="https://i.ytimg.com/vi/hY7m5jjJ9mM/default.jpg"
          tabIndex="0"
        />
      </div>
      <div
        className="panel grid-item"
      >
        <div
          className="play"
        >
          <button
            className="buttonPlay"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="glyphicon glyphicon-play-circle"
            />
          </button>
        </div>
        <div
          className="delete"
        >
          <button
            className="buttonDelete"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="glyphicon glyphicon-trash"
            />
          </button>
        </div>
        <div
          className="addFav"
        >
          <button
            className="buttonAddFav"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="glyphicon glyphicon-star-empty"
            />
          </button>
        </div>
      </div>
      <div
        className="info"
      >
        <div
          className="title infobox"
        >
          CATS will make you LAUGH YOUR HEAD OFF - Funny CAT compilation
        </div>
        <div
          className="addedAt infobox"
        >
          Wed May 31 2017
        </div>
        <div
          className="likes infobox"
        >
          <span>
            327046
          </span>
           
          <span
            className="glyphicon glyphicon-heart"
          />
        </div>
        <div
          className="views infobox"
        >
          <span>
            81007082
          </span>
           views
        </div>
      </div>
    </div>
    <div
      className="grid-container videocontainer"
    >
      <div
        className="pic grid-item"
      >
        <img
          alt="thumbnail"
          onClick={[Function]}
          src="https://i.ytimg.com/vi/mRf3-JkwqfU/default.jpg"
          tabIndex="0"
        />
      </div>
      <div
        className="panel grid-item"
      >
        <div
          className="play"
        >
          <button
            className="buttonPlay"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="glyphicon glyphicon-play-circle"
            />
          </button>
        </div>
        <div
          className="delete"
        >
          <button
            className="buttonDelete"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="glyphicon glyphicon-trash"
            />
          </button>
        </div>
        <div
          className="addFav"
        >
          <button
            className="buttonAddFav"
            onClick={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="glyphicon glyphicon-star-empty"
            />
          </button>
        </div>
      </div>
      <div
        className="info"
      >
        <div
          className="title infobox"
        >
          Funny Puppies And Cute Puppy Videos Compilation 2016 [BEST OF]
        </div>
        <div
          className="addedAt infobox"
        >
          Wed Aug 31 2016
        </div>
        <div
          className="likes infobox"
        >
          <span>
            71389
          </span>
           
          <span
            className="glyphicon glyphicon-heart"
          />
        </div>
        <div
          className="views infobox"
        >
          <span>
            17981823
          </span>
           views
        </div>
      </div>
    </div>
  </div>
</div>
`;
